Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContext.h

```diff
@@ -57,21 +56,43 @@ class MediaPlaybackTargetContext {
     MediaPlaybackTargetContext() = default;
     WEBCORE_EXPORT explicit MediaPlaybackTargetContext(RetainPtr<AVOutputContext>&&);
 
-    MediaPlaybackTargetContext(RetainPtr<NSData>&& serializedOutputContext, bool hasActiveRoute)
-        : m_type(Type::SerializedAVOutputContext)
-        , m_serializedOutputContext(WTFMove(serializedOutputContext))
-        , m_cachedHasActiveRoute(hasActiveRoute)
-    {
-        ASSERT(m_serializedOutputContext);
-    }
-
     MediaPlaybackTargetContext(const String& mockDeviceName, MockState state)
         : m_type(Type::Mock)
         , m_mockDeviceName(mockDeviceName)
         , m_mockState(state)
     {
     }
 
+    struct NonPlatformData {
+        String mockDeviceName;
+        MockState mockState;
+    };
+
+    struct PlatformData {
+        RetainPtr<AVOutputContext> outputContext;
+    };
+
+    using PlatformDataVariant = std::variant<PlatformData, NonPlatformData>;
+
+    MediaPlaybackTargetContext(bool hasPlatformData, Type type, std::optional<PlatformDataVariant>&& platformDataVariant)
+        : m_hasPlatformData(hasPlatformData)
+        , m_type(type)
+        {
+            if (platformDataVariant) {
```

Comment:
I think we need to fail decoding if this is false, in order to maintain pre-existing behavior. I'd suggest we use `[CreateUsing=fromIPCData]` in the serialization.in. Then you replace this constructor with a static function named `fromIPCData` which returns a `std::optional< MediaPlaybackTargetContext>`. In case of failure (e.g. `platformDataVariant` is `std::nullopt`, have the function return std::nullopt so that decoding fails.